test/test_nn.py

```diff
@@ -5062,20 +5062,20 @@ def test_hardtanh_backward(self):
         self.assertEqual(x.grad, x_grad_ref)
 
     def test_batchnorm_nhwc_cpu(self):
-        def helper(self, size, dtype, mixed_dtype=False):
+        def helper(self, mod, size, dtype, mixed_dtype=False, format=torch.channels_last):
```

Comment:
> the effect it achieves is similar to that of this test except for
1. the functionality of the mixed_dtype arg
2. the checks on backward

Yes.

> BatchNorm3d (which has 5d sample inputs) does not seem to have supported channels_last_3d prior to this PR (I don't see the test being xfailed in the ModuleInfo)

Actually, BatchNorm3d can not utilize channels_last_3d kernel prior to this PR but it can has channels_last_3d support using another code path by TensorIterator https://github.com/pytorch/pytorch/pull/97774/files#diff-9e36531b6ea57776251e04cf2e1b84a94145020d870fed3d7f82388c47371483L177.

Reply:
Ah I see, so this PR allows BatchNorm3d to use a fast kernel for `channels_last_3d` inputs without transforming the inputs, is that correct?

Would you be open to updating [test_memory_format](https://github.com/pytorch/pytorch/blob/main/test/test_modules.py#L604)? I don't want to block this PR if the change is urgent but it would be some super nice cleanup since it seems like you are updating other functions to support `channels_last_3d` as well and could catch cases where backward kernels are not properly supporting `channels_last_3d` :)

Reply:
Yes, I'm willing to add checks on backward for `test_memory_format`.